src/Testcontainers/Configurations/WaitStrategies/HttpWaitStrategy.cs

```diff
@@ -0,0 +1,212 @@
+﻿namespace DotNet.Testcontainers.Configurations
+{
+  using System;
+  using System.Collections.Generic;
+  using System.Linq;
+  using System.Net;
+  using System.Net.Http;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Containers;
+  using JetBrains.Annotations;
+  using Microsoft.Extensions.Logging;
+
+  /// <summary>
+  /// Wait for an HTTP(S) endpoint to return a particular status code.
+  /// </summary>
+  [PublicAPI]
+  public sealed class HttpWaitStrategy : IWaitUntil
+  {
+    private const ushort HttpPort = 80;
+
+    private const ushort HttpsPort = 443;
+
+    private readonly IDictionary<string, string> httpHeaders = new Dictionary<string, string>();
+
+    private readonly ISet<HttpStatusCode> httpStatusCodes = new HashSet<HttpStatusCode>();
+
+    private Predicate<HttpStatusCode> httpStatusCodePredicate;
+
+    private HttpMethod httpMethod;
+
+    private string schemeName;
+
+    private string pathValue;
+
+    private ushort? portNumber;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HttpWaitStrategy" /> class.
+    /// </summary>
+    public HttpWaitStrategy()
+    {
+      _ = this.WithMethod(HttpMethod.Get).UsingTls(false).ForPath("/");
+    }
+
+    /// <inheritdoc />
+    public async Task<bool> Until(ITestcontainersContainer testcontainers, ILogger logger)
+    {
+      // Java fall back to the first exposed port. The .NET wait strategies do not have access to the exposed port information yet.
+      var containerPort = this.portNumber.GetValueOrDefault(Uri.UriSchemeHttp.Equals(this.schemeName, StringComparison.OrdinalIgnoreCase) ? HttpPort : HttpsPort);
```

Comment:
@eddumelendez @kiview I've choosen this fall back until the .NET wait strategies can access the expoed ports.

Reply:
Sounds good for the first implementation 👍